### PR TITLE
Add native versions of chop, chomp, first, and last

### LINKED ISSUE #17: views of InlineStrings
Currently e.g. we have:
```
julia> using InlineStrings

julia> x = InlineString("abcd")
"abcd"

julia> y = last(x, 3)
"bcd"

julia> typeof(x)
String7

julia> typeof(y)
SubString{String7}

julia> Base.summarysize(x)
8

julia> Base.summarysize(y)
24
```
which seems wasteful. Maybe for small inline strings we should have special implementation of standard functions like `first`, `last`, `chop` that would not make a view, but just create a new string of the same type as the source?

### PR DESCRIPTION
**Pull request by quinnj**
Fixes #17. Allows returning inline string types from inline string type
inputs.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -122,6 +122,24 @@ end # @testset
...
+        @test chop(x; head=sizeof(x) + 1) == chop(y; head=sizeof(y) + 1)
+        @test chop(x; head=sizeof(x) + 1, tail=sizeof(x) + 1) == chop(y; head=sizeof(x) + 1, tail=sizeof(y) + 1)
+        y != "" && @test first(x) == first(y)
+        @test first(x, sizeof(x)) == first(y, sizeof(y))
+        y != "" && @test first(x, sizeof(x) - 1) == first(y, sizeof(y) - 1)
+        @test first(x, sizeof(x) + 1) == first(y, sizeof(y) + 1)
+        y != "" && @test last(x) == last(y)
+        @test last(x, sizeof(x)) == last(y, sizeof(y))
+        y != "" && @test last(x, sizeof(x) - 1) == last(y, sizeof(y) - 1)
+        @test last(x, sizeof(x) + 1) == last(y, sizeof(y) + 1)
```
Comment (bkamins) [inline test/runtests.jl:142]: It would to have tests for `n` equal to 0 and -1 for `last` and `first`.

Comment (quinnj) [reply test/runtests.jl:142]: Yeah, 0 would be good. I did check -1, but it throws an error in `nextind`/`prevind` for both inline strings/strings, so I figured that was fine.

Comment (bkamins) [reply test/runtests.jl:142]: yes - for -1 it should throw this error, I just meant to check it in case in the future implementation changed and did not call these internally.

File: test/runtests.jl [hunk 2]
```diff
@@ -122,6 +122,24 @@ end # @testset
         write(io, x)
         seekstart(io)
         @test read(io, typeof(x)) === x
+        @test chomp(x) == chomp(y)
+        if typeof(x) != String255
+            @test chomp(InlineString(x * "\n")) == chomp(y * "\n")
+            @test chomp(InlineString(x * "\r\n")) == chomp(y * "\r\n")
+        end
+        @test chop(x) == chop(y)
```
Comment (bkamins) [inline test/runtests.jl:130]: also here maybe testing `head=0` and `tail=0`? (again negative indices should be caught by `nextind` and `prevind`)

**Comment by bkamins** [comment]
I have tested the cases I asked about locally and they pass correctly.
